### ISSUE TITLE
Featured Product block breaks when product is out of stock + hidden from catalog

### ISSUE THREAD

**Issue by cuemarie** [Automattic/wp-calypso#58595]
### Quick summary

'Featured Product' block breaks in the editor when the product is out of stock _and_ the Inventory is set to hide out of stock items from the catalog. Selecting `edit` results in `This block has encountered an error and cannot be previewed.`

### Steps to reproduce

1. Open a WooCommerce Site
2. Add a Featured Product block to a page and save
3. Head to the products and set the product to **Out of Stock**
4. Under WooCommerce > Settings > Products > Inventory, check the box that says "Hide out of stock items from the catalog"
5. Return to the page with your Featured Product block, select it, and select "Edit" to choose a new product to feature

<img width="960" alt="Markup 2021-11-29 at 10 04 32" src="https://user-images.githubusercontent.com/27249804/143914067-7ce7d817-1d57-4575-8045-c732219e9d46.png">

<img width="1095" alt="Markup 2021-11-29 at 10 11 34" src="https://user-images.githubusercontent.com/27249804/143914084-30fc0cf8-319c-44e7-ab74-d4f28237a64a.png">

### What you expected to happen

The block can still be edited to choose a new product. 

- When "Hide out of stock items from the catalog" is _unchecked_, the block responds as expected. 

### What actually happened

Clicking edit results in a broken block

https://user-images.githubusercontent.com/27249804/143914004-d40f04aa-967b-4c61-9be7-57e00d209d06.mp4

### Context

Reported by a user here: 32966334-hc
Was able to recreate on my test site

### Operating System

macOS

### Browser

Mozilla Firefox

### Simple, Atomic or both?

Atomic

### Reproducibility

Consistent

### Severity

Some (< 50%)

### Available workarounds?

Yes, easy to implement

### Workaround details

Removing the block and re-inserting it is a quick workaround

**Comment by kosiew** [comment]
Hi @cuemarie ,

Thanks for reporting this and the detailed steps.

I could not reproduce this on https://wpstore101.wpcomstaging.com/wp-admin/post.php?post=6208&action=edit

Can you check further on how I can reproduce it?

**Comment by cuemarie** [comment]
Hey @kosiew ! I'm still able to reproduce the issue on my end. Should I be able to access that `wpstore101` link? I currently can't, but perhaps I'm missing a step there. :) 

Just recreated the issue with a second product on my test site by:

1. Adding a Featured Product block to a page, and selecting an **in stock** product, saving changes
2. Double checked that WooCommerce > Settings > Products > Inventory has `Hide out of stock items from the catalog` **checked**
3. Went to All Products, selected **Quick Edit** for the product I chose in step 1, and set the status to **Out of Stock**
4. Returned to the page I used in step one, selected the Featured Product block, and selected the **edit** pencil. This produced the same result as before: 

<img width="864" alt="Screen Shot 2021-12-02 at 10 07 03 AM" src="https://user-images.githubusercontent.com/27249804/144469313-43ad7cb1-6056-4e0b-849b-4497125712ba.png">
